arkouda/dataframe.py

```diff
@@ -1845,6 +1848,261 @@ def numeric_help(d):
+
+        self.name = None  # Clear our internal DataFrame object name
+
+    def is_registered(self) -> bool:
+        """
+        Return True if the object is contained in the registry
+
+        Returns
+        -------
+        bool
+            Indicates if the object is contained in the registry
+
+        Raises
+        ------
+        RegistrationError
+            Raised if there's a server-side error or a mismatch of registered components
+
+        See Also
+        --------
+        register, attach, unregister, unregister_dataframe_by_name
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+
+        if self.name is None:
+            return False
+
+        # Total number of registered parts should equal number of columns plus an entry
+        # for the index and the column order
+        total = len(self.data) + 2
+        registered = 0
+
+        for col, data in self.data.items():
+            if data.is_registered():
+                registered += 1
+
+        if self.index.values.is_registered():
+            registered += 1
+        if f"df_columns_{self.name}" in list_registry():
+            registered += 1
+
+        if 0 < registered < total:
+            warn(
+                f"WARNING: DataFrame {self.name} expected {total} components to be registered,"
+                f" but only located {registered}."
+            )
+
+        return registered == total
+
+    @staticmethod
+    def attach(user_defined_name: str) -> DataFrame:
+        """
+        Function to return a DataFrame object attached to the registered name in the
+        arkouda server which was registered using register()
+
+        Parameters
+        ----------
+        user_defined_name : str
+            user defined name which DataFrame object was registered under
+
+        Returns
+        -------
+        DataFrame
+               The DataFrame object created by re-attaching to the corresponding server components
+
+        Raises
+        ------
+        RegistrationError
+            if user_defined_name is not registered
+
+        See Also
+        --------
+        register, is_registered, unregister, unregister_groupby_by_name
+        """
+
+        col_resp = cast(str, generic_msg(cmd="stringsToJSON", args=f"df_columns_{user_defined_name}"))
+        columns = dict.fromkeys(json.loads(col_resp))
+        matches: List[str] = []
+        regEx = compile(
+            f"^df_data_[a-zA-Z0-9]+_({pdarray.objtype}|{Strings.objtype}|"
+            f"{Categorical.objtype}|{SegArray.objtype})_{user_defined_name}"
+        )
+        # Using the regex, cycle through the registered items and find all the columns in the DataFrame
+        for name in list_registry():
+            x = match(regEx, name)
+            if x is not None:
+                matches.append(x.group())
+        matches.sort()
```

Comment:
we could do something like this
```python
matches = builtins.sorted([x.group() for name in list_registry() if (x := match(regEx, name)) is not None])
```
which would eliminate the need for explicitly typing `matches` above

Note this requires us to `import builtins` because at some point in the past we overrode `sorted` from `builtins`... which I don't love

Reply:
Sounds good! The overriding then `__builtin__.sorted` has been captured in https://github.com/Bears-R-Us/arkouda/issues/1734